cli/pkg/config/cli_config.go

```diff
@@ -1,39 +1,57 @@
 package config
 
 import (
+	"bytes"
 	"encoding/json"
 	"fmt"
-	"github.com/keptn/go-utils/pkg/common/fileutils"
-	"io/ioutil"
 	"log"
 	"time"
 
+	"github.com/keptn/go-utils/pkg/common/fileutils"
+	"github.com/mitchellh/mapstructure"
+	"github.com/spf13/viper"
+
 	keptnutils "github.com/keptn/kubernetes-utils/pkg"
 )
 
 // CLIConfig holds infos of the CLI config
 type CLIConfig struct {
-	AutomaticVersionCheck bool       `json:"automatic_version_check"`
-	KubeContextCheck      bool       `json:"kube_context_check"`
-	LastVersionCheck      *time.Time `json:"last_version_check"`
-	CurrentContext        string     `json:"current-context"`
+	AutomaticVersionCheck bool       `json:"automatic_version_check" mapstructure:"automatic_version_check"`
+	KubeContextCheck      bool       `json:"kube_context_check" mapstructure:"kube_context_check"`
+	LastVersionCheck      *time.Time `json:"last_version_check" mapstructure:"last_version_check"`
+	CurrentContext        string     `json:"current-context" mapstructure:"current-context"`
 }
 
 // CLIConfigManager manages the path of the CLI config
 type CLIConfigManager struct {
 	CLIConfigPath string
+	viper         *viper.Viper
 }
 
+var mgr *CLIConfigManager
+
 // NewCLIConfigManager creates a new CLIConfigManager
-func NewCLIConfigManager() *CLIConfigManager {
-	cliConfigManager := CLIConfigManager{}
+func NewCLIConfigManager(cfgFile string) *CLIConfigManager {
+	if mgr != nil {
+		return mgr
+	}
+	if cfgFile == "" {
+		var err error
+		cfgFile, err = GetKeptnDefaultConfigPath()
+		if err != nil {
+			log.Fatal("could not get default config path", err)
+		}
 
-	dir, err := keptnutils.GetKeptnDirectory()
-	if err != nil {
-		log.Fatal(err)
+		fmt.Printf("no value for --config-file was specified\n defaulting to: %s\n", cfgFile)
```

Comment:
please use `logging.PrintLog()` with `VerboseLevel` instead of `fmt.Printf`

Reply:
I will change this. Thanks for the review :pray: